core/bin/snapshots_creator/src/main.rs

```diff
@@ -136,15 +142,19 @@ async fn run(
     master_pool: ConnectionPool,
 ) -> anyhow::Result<()> {
     let latency = METRICS.snapshot_generation_duration.start();
-
     let config = SnapshotsCreatorConfig::from_env().context("SnapshotsCreatorConfig::from_env")?;
 
     let mut conn = replica_pool
         .access_storage_tagged("snapshots_creator")
         .await?;
 
-    // we subtract 1 so that after restore, EN node has at least one l1 batch to fetch
-    let l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await? - 1;
+    // We subtract 1 so that after restore, EN node has at least one L1 batch to fetch
+    let sealed_l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await?;
+    if sealed_l1_batch_number == L1BatchNumber(0) {
+        tracing::info!("Node has no sealed L1 batches other than the genesis one; exiting");
+        return Ok(());
```

Comment:
IMO, it would be inconsistent with the behavior when there's already a snapshot for the L1 batch; I'd consider the genesis L1 batch a trivial case of a snapshot.